### PR TITLE
Make DEFAULT_CONTROLS and ReactionPredicate.*_EMOJIS immutable

### LINKED ISSUE #5586: Make DEFAULT_CONTROLS and ReactionPredicate.*_EMOJIS immutable
This ensures that developers don't accidentally mutate these while working with them.

### PR DESCRIPTION
**Pull request by Jackenmen**
### Description of the changes

Fixes #5586 
I also improved typings for stuff in `redbot.core.utils.menus` because I already had to update them to be compatible with MappingProxyType anyway...

### Have the changes in this PR been tested?

No

### CODE REVIEW

File: redbot/core/utils/predicates.py [hunk 1]
```diff
@@ -881,19 +881,19 @@ class ReactionPredicate(Callable[[discord.Reaction, discord.abc.User], bool]):
...
             ord("\N{REGIONAL INDICATOR SYMBOL LETTER A}"),
             ord("\N{REGIONAL INDICATOR SYMBOL LETTER Z}") + 1,
         )
-    ]
-    """List[str] : A list of all 26 alphabetical letter emojis."""
+    )
+    """Tuple[str, ...] : A tuple of all 26 alphabetical letter emojis."""
 
-    NUMBER_EMOJIS: ClassVar[List[str]] = [
+    NUMBER_EMOJIS: ClassVar[Tuple[str, ...]] = [
```
Comment (TrustyJAID) [inline redbot/core/utils/predicates.py]: You didn't actually convert this one to a tuple

**Comment by TrustyJAID** [comment]
I am not sure how I feel about this one. Making DEFAULT_CONTROLS immutable adds more work to the end user wanting to add additional buttons to the default controls. With this they would have to import each individual control to build a new mapping of functionality instead of just adding new methods to the default controls or iterate through the MappingProxyType and build a new one with additional controls on it.

**Comment by Jackenmen** [comment]
> I am not sure how I feel about this one. Making DEFAULT_CONTROLS immutable adds more work to the end user wanting to add additional buttons to the default controls.

You just need to replace usage of `DEFAULT_CONTROLS` with `dict(DEFAULT_CONTROLS)` and making the object immutable actually prevents the user from accidentally overriding the default controls dictionary (that is used across the whole) when they try to use `DEFAULT_CONTROLS` + some custom control or something like that.

**Review by TrustyJAID — APPROVED** [review]
Looks good to me!

**Pull request merged**